src/main/kotlin/cacophony/grammar/syntaxtree/Expression.kt

```diff
@@ -0,0 +1,279 @@
+package cacophony.grammar.syntaxtree
+
+import cacophony.utils.Location
+
+sealed class Expression(val range: Pair<Location, Location>) { // everything in cacophony is an expression
+    // artificial instance, can be useful when calculating values of nested expressions
+    class EmptyExpression(
+        range: Pair<Location, Location>,
+    ) : Expression(range)
+
+    class Variable(
+        range: Pair<Location, Location>,
+    ) : Expression(range)
+
+    // should Type also inherit expression? if so, can we do for example "; Bool;" in our language?
+    class Type(
+        range: Pair<Location, Location>,
+    ) : Expression(range)
+
+    class VariableDeclaration(
+        range: Pair<Location, Location>,
+        val keywordLet: Keyword.KeywordLet,
+    ) : Expression(range)
+
+    class FunctionArgument(
+        range: Pair<Location, Location>,
+        type: Type,
+    ) : Expression(range)
+
+    class VariableTypedDeclaration(
+        range: Pair<Location, Location>,
+        variable: VariableDeclaration,
+        type: Type,
+    ) : Expression(range)
+
+    class Literal(
+        range: Pair<Location, Location>,
```

Comment:
**suggestion:** As for `Variable` and `Type`, we could store the value of the literal. Also it might be good to distinguish between different types of literals here, i.e. something like `BooleanLiteral` (or even `True`, `False`), `IntegerLiteral`

Reply:
then maybe we should distinguish them at lexer level?

Reply:
I think that would probably be the best